modules: hal_nordic: 802.15.4 carrier functions made optional

ahasztag opened the pull request:
This function adds an option to make the use of continuous carrier and modulated carrier functions optional in the Nordic 802.15.4 driver. This allows to save some memory in applications which do not use these functions

ahasztag committed:
modules: hal_nordic: 802.15.4 carrier functions made optional

Added an option to disable and enable the continuous and optional
carrier functions in the nordic 802.15.4 driver.

Signed-off-by: Artur Hadasz <artur.hadasz@nordicsemi.no>

jciupis commented on modules/hal_nordic/Kconfig:
```diff
@@ -205,6 +205,17 @@ endif
+
+config NRF_802154_CARRIER_FUNCTIONS
+	bool "Enable 802.15.4 carrier functions"
+	default n
```
No need to explicitly set it: https://docs.zephyrproject.org/1.14.0/guides/kconfig/index.html#redundant-defaults.

jciupis commented on modules/hal_nordic/nrf_802154/CMakeLists.txt:
```diff
@@ -94,6 +94,8 @@ set(NRF53_SERIES  ${CONFIG_SOC_SERIES_NRF53X})
 set(SER_HOST      ${CONFIG_NRF_802154_SER_HOST})
 set(SL_OPENSOURCE ${CONFIG_NRF_802154_SL_OPENSOURCE})
 
+set(NRF_802154_CARRIER_FUNCTIONS_ENABLED ${CONFIG_NRF_802154_CARRIER_FUNCTIONS})
```
Shouldn't it be done using `target_compile_definitions` syntax like other configuration macros above?

jciupis commented on modules/hal_nordic/Kconfig:
```diff
@@ -205,6 +205,16 @@ endif
+	bool "Enable 802.15.4 carrier functions"
+	help
+	  This option enables functions such as modulated carrier and continuous carrier.
+	  If this option is modified, its remote counterpart must be set to the exact same value.
```
`If this option is modified on a multicore SoC, ...`

Otherwise this documentation is misleading for nRF52 users.

ahasztag replied:
Please check it now,

hubertmis commented on modules/hal_nordic/Kconfig:
```diff
@@ -205,6 +205,16 @@ endif
 
 endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
 
+if (NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION)
```
nit: `depends on` could be cleaner than the `if` - `endif` block

ahasztag replied:
Ok, changed

rlubos reviewed (dismissed):
@ahasztag Please address the compliance warning:
>  Boolean option 'NRF_802154_CARRIER_FUNCTIONS' prompt must not start with 'Enable...'. Please
check Kconfig guidelines.
 Boolean option 'NRF_802154_CARRIER_FUNCTIONS' prompt must not start with 'Enable...'. Please
check Kconfig guidelines.

Otherwise, LGTM.

ahasztag committed:
manifest: update hal_nordic revision

This commit updates hal_nordic revision to bring the latest changes in
nRF IEEE 802.15.4 radio driver.

Signed-off-by: Artur Hadasz <artur.hadasz@nordicsemi.no>

rlubos commented on west.yml:
```diff
@@ -83,7 +83,7 @@ manifest:
         - hal
     - name: hal_nordic
-      revision: 3f7ae2320b6d348898c7b11e956a6e792d0b3724
+      revision: pull/125/head
```
Please update SHA: `6c9f23498ee6a225fee049c8c5b2e1ac760bd4a8`

And please, fix the compliance.

ahasztag replied:
Should be ok now.

ahasztag commented on modules/hal_nordic/nrf_802154/CMakeLists.txt:
```diff
@@ -89,6 +89,10 @@ else()
 
+if (CONFIG_NRF_802154_CARRIER_FUNCTIONS)
+  target_compile_definitions(zephyr-802154-interface INTERFACE NRF_802154_CARRIER_FUNCTIONS_ENABLED=1)
+endif()
```
Ok, went with the "else" option, as introducing the change into `nrf_802154_config.h` would be pretty complicated and would require opening a lot of PR-s. You can take a look now.

maciejbaczmanski reviewed (approved):
LGTM. Thanks
